File: 프로그래머스/unrated/181923. 수열과 구간 쿼리 2/수열과 구간 쿼리 2.java
import java.util.*;

class Solution {
    public int[] solution(int[] arr, int[][] queries) {
        int[] result = new int[queries.length];

        int i = 0;
        for (int[] query : queries) {
            int s = query[0];
            int e = query[1];
            int k = query[2];

            int minGreater = Integer.MAX_VALUE;
            boolean found = false;

            for (int j = s; j <= e; j++) {
                if (arr[j] > k && arr[j] < minGreater) {
                    minGreater = arr[j];
                    found = true;
                }
            }

            result[i++] = found ? minGreater : -1;
        }

        return result;
    }
}

File: 프로그래머스/0/181923. 수열과 구간 쿼리 2/수열과 구간 쿼리 2.py
def solution(arr, queries):
    answer = []

    for s, e, k in queries:
        valid_values = [value for value in arr[s:e+1] if value > k]

        if valid_values:
            answer.append(min(valid_values))
        else:
            answer.append(-1)

    return answer
